src/core/send.c

```diff
@@ -1018,6 +1018,45 @@ QuicSendGetNextStream(
     return NULL;
 }
 
+BOOLEAN
+CxPlatIsRouteReady(
+    _In_ QUIC_CONNECTION *Connection,
+    _In_ BOOLEAN PathChallenge
+) {
+    QUIC_PATH* Path = &Connection->Paths[0];
+    //
+    // Make sure the route is resolved before sending packets.
+    //
+    //
+    // We need to set the path challenge flag back on so that when route is resolved,
+    // we know we need to continue to send the challenge.
+    //
+    CXPLAT_DBG_ASSERT((!PathChallenge && Path->IsActive) ||
+                      (PathChallenge && Path->Route.State != RouteSuspected));
+    if ((!PathChallenge && (Path->Route.State == RouteUnresolved || Path->Route.State == RouteSuspected)) ||
```

Comment:
@csujedihy & @maolson-msft do we really need special logic for "Path Challenge" scenarios where only `RouteUnresolved` will trigger route resolution, instead of also `RouteSuspected`?

Reply:
I think I added that path challenge route resolution logic. I think the assumption I had is when sending path challenge, it's impossible that the route on the path is in suspected state because suspected state is only valid if we already have the route resolved. But, path challenge can only happen on a fresh path that either has not resolved the route or is in middle of resolving it.

Reply:
And that assert has not been hit so far. So, I think my assumption is still correct.

Reply:
In the future, we might send a challenge on a previously valid, but now suspect path. We don't today, but shouldn't preclude the possibility. Do you see any reason not to have the same logic for normal sends and path challenges?

Reply:
No, it's probably fine to remove that suspected check. At least we can try.